salt: fix urllib.parse module

SpiritLooper opened the pull request:
Retrograding python package version from 3.12 to 3.10.

Since the upgrade to version 24.11, Salt has exhibited strange behavior with SLS files:
- https://github.com/saltstack/salt/issues/67097#issuecomment-2558397227
- https://discourse.nixos.org/t/upgrade-to-24-11-gives-issue-in-salt/57735

In salt’s GitHub workflow, we can see that salt has been tested with Python 3.10.
Retrograding python to the tested version resolves the issues. 

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/upgrade-to-24-11-gives-issue-in-salt/57735/2

Flakebi reviewed (approved):
Looks good to me, thanks.
Looks like there is a fix in progress upstream: https://github.com/saltstack/salt/pull/66899

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 372604`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.pytest-testinfra</li>
    <li>python312Packages.pytest-testinfra.dist</li>
    <li>salt</li>
    <li>salt.dist</li>
  </ul>
</details>

mweinelt commented:
Please apply then patch and stick with the default Python version.

SpiritLooper committed:
salt: fix urllib.parse module

Patch applied from Fedora
https://src.fedoraproject.org/rpms/salt/blob/1c6e7b7a88fb81902f5fcee32e04fa80713b81f8/f/urllib.patch

SpiritLooper commented:
As @mweinelt advised, I applied the patch from Fedora mainstream and revert my previous commit. 
I ran tests on my side and the patch seems to work :partying_face:

Flakebi commented:
I get a hash-mismatch for the patch:
```
error: hash mismatch in fixed-output derivation '/nix/store/brx3x26n6zh0jw19l557c2hw5z13c063-urllib.patch.drv':
        likely URL: https://src.fedoraproject.org/rpms/salt/raw/1c6e7b7a88fb81902f5fcee32e04fa80713b81f8/f/urllib.patch
         specified: sha256-+asvGPv4XIoOu6aqiP5BX5k8o3eqoPNUGzLSnw1pDFo=
            got:    sha256-yldIurafduOAYpf2X0PcTQyyNjz5KKl/N7J2OTEF/c0=
```

Flakebi reviewed (approved):
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 372604`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.pytest-testinfra</li>
    <li>python312Packages.pytest-testinfra.dist</li>
    <li>salt</li>
    <li>salt.dist</li>
  </ul>
</details>
